torch/_inductor/freezing.py

```diff
@@ -241,3 +245,60 @@ def discard_traced_gm_params(mod):
             e_t.requires_grad_(True)
             e_t._is_param = True
         setattr(mod, attr_name, e_t)
+
+
+def enforce_output_layout(gm):
+    """
+    Make sure the output node's layout does not change due to compiler optimizations
+    by adding aten.as_strided nodes with the expected strides.
+
+    Only used for inference so we can assume all graph outputs are model outputs.
+    """
+    *_, output_node = gm.graph.nodes
+    out_list, spec = pytree.tree_flatten(output_node.args)
+    with gm.graph.inserting_before(output_node):
+        for n in out_list:
+            if not isinstance(
+                n.meta["val"], torch.Tensor
+            ) or not torch._prims_common.is_non_overlapping_and_dense(n.meta["val"]):
+                continue
+
+            # add a node to enforce eager layout
+            ft = n.meta["val"]
+            new_node = gm.graph.call_function(
```

Comment:
Hm, is this very rare that we have a single tensor appear multiple times in the output?

I can add the cache if that happens common enough. Otherwise the next day when I or someone else reads the code, people may wondering why such a cache is necessary.